subsystem/cluster_test.go

```diff
@@ -2805,6 +2808,175 @@ var _ = Describe("cluster install, with default network params", func() {
+			RAMMib:     8192,
+		}
+		workerCNVRequirements = models.ClusterHostRequirementsDetails{
+			CPUCores: 2,
+			RAMMib:   360,
+		}
+		masterCNVRequirements = models.ClusterHostRequirementsDetails{
+			CPUCores: 4,
+			RAMMib:   150,
+		}
+		workerTotalRequirements = models.ClusterHostRequirementsDetails{
+			CPUCores:   workerOCPRequirements.CPUCores + workerCNVRequirements.CPUCores,
+			RAMMib:     workerOCPRequirements.RAMMib + workerCNVRequirements.RAMMib,
+			DiskSizeGb: workerOCPRequirements.DiskSizeGb,
+		}
+		masterTotalRequirements = models.ClusterHostRequirementsDetails{
+			CPUCores:   masterOCPRequirements.CPUCores + masterCNVRequirements.CPUCores,
+			RAMMib:     masterOCPRequirements.RAMMib + masterCNVRequirements.RAMMib,
+			DiskSizeGb: masterOCPRequirements.DiskSizeGb,
+		}
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		registerClusterReply, err := userBMClient.Installer.RegisterCluster(ctx, &installer.RegisterClusterParams{
+			NewClusterParams: &models.ClusterCreateParams{
+				BaseDNSDomain:            "example.com",
+				ClusterNetworkCidr:       &clusterCIDR,
+				ClusterNetworkHostPrefix: 23,
+				Name:                     swag.String("test-cluster"),
+				OpenshiftVersion:         swag.String(common.TestDefaultConfig.OpenShiftVersion),
+				PullSecret:               swag.String(pullSecret),
+				ServiceNetworkCidr:       &serviceCIDR,
+				SSHPublicKey:             sshPublicKey,
+			},
+		})
+		Expect(err).ToNot(HaveOccurred())
+		cluster = registerClusterReply.GetPayload()
+		clusterID = *cluster.ID
+
+		registerHostsAndSetRoles(clusterID, 5)
+
+		clusterResp, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{
+			ClusterID: clusterID,
+		})
+		Expect(err).ToNot(HaveOccurred())
+		cluster = clusterResp.GetPayload()
+	})
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	It("should be reported for cluster without operators", func() {
+		params := installer.GetClusterHostRequirementsParams{
+			ClusterID: clusterID,
+		}
+
+		response, err := userBMClient.Installer.GetClusterHostRequirements(ctx, &params)
+
+		Expect(err).ToNot(HaveOccurred())
+		requirements := response.GetPayload()
+		hosts := cluster.Hosts
+		Expect(requirements).To(HaveLen(len(hosts)))
+
+		hostIDToRequirements := make(map[strfmt.UUID]*models.ClusterHostRequirements)
+		for _, rq := range requirements {
+			hostIDToRequirements[rq.HostID] = rq
+		}
+
+		for _, host := range hosts {
+			hostRequirements := hostIDToRequirements[*host.ID]
+			expectedRequirements := workerOCPRequirements
+			if host.Role == models.HostRoleMaster {
+				expectedRequirements = masterOCPRequirements
+			}
+			Expect(hostRequirements.HostID).To(Equal(*host.ID))
+			Expect(hostRequirements.Operators).To(BeEmpty())
+
+			Expect(*hostRequirements.Ocp).To(BeEquivalentTo(expectedRequirements))
+			Expect(hostRequirements.Total).To(BeEquivalentTo(hostRequirements.Ocp))
+		}
+	})
+
+	It("should be reported for cluster with operators", func() {
```

Comment:
This test has a lot of lines. Try to split it into an helper function to be used in both tests.